lib/facade/MonitoringFacade.ts

```diff
@@ -310,11 +357,23 @@ export class MonitoringFacade extends MonitoringScope {
       .map((s) => s as Monitoring);
   }
 
+  /**
+   * Adds a dashboard segment which returns dynamic content depending on dashboard type.
+   * @param segment dynamic segment to add.
+   */
+  addDynamicSegment(segment: IDynamicDashboardSegment) {
+    const dashboardFactory = this.dashboardFactory as IDynamicDashboardFactory;
```

Comment:
Shouldn't there be a check here too?

I wonder if we should just deprecate these at the facade-level and make it so callers would call something like `facade.dashboardFactory.addDynamicSegment(...)`, if typing is good enough for that.

Reply:
In my new change, I've deprecated the previous `addSegment` method and it now delegates to the `addDynamicSegment` method using an adapter.